src/mbgl/map/tile_worker.hpp

```diff
@@ -31,7 +31,7 @@ class GeometryTileLayer;
 class TileParseResultBuckets {
 public:
     TileData::State state = TileData::State::invalid;
-    std::vector<std::pair<std::string, std::unique_ptr<Bucket>>> buckets;
+    std::unordered_map<std::string, std::unique_ptr<Bucket>> buckets;
```

Comment:
I changed this so that I could do `buckets = std::move(resultBuckets.buckets);` on the `VectorTileData`-side result of `parseAllLayers`. In the "all layers" case, we want to remove any old buckets that no longer exist (e.g., removed annotations).